uber/models.py

```diff
@@ -2086,7 +2089,7 @@ def register_session_listeners():
     """
     listen(Session.session_factory, 'before_flush', _acquire_badge_lock)
     listen(Session.session_factory, 'before_flush', _presave_adjustments)
-    listen(Session.session_factory, 'before_flush', _track_changes)
+    listen(Session.session_factory, 'after_flush', _track_changes)
```

Comment:
This is the line. When we save to the tracking table after the flush, we can still inspect the session for dirty objects, but we get the new foreign keys that we were missing before.

@EliAndrewC says he'll look into it extra to make double triple sure that this is a good and safe change, but the likely scenario is that we had it as `before_flush` because `after_flush` implies that you can't inspect the session for dirty objects (and we thus wouldn't be able to tell what actually changed when an object updated). But it doesn't. That's `after_flush_postexec()`. Oh, SQLAlchemy, :\